### DISCUSSION TITLE
How do you bind attributes of tags to state vars changes

### DISCUSSION THREAD

**Discussion by iongion** [vanjs-org/van#22] [Q&A] [answered]
By intuition I tried this and is not working

```typescript
const { div, img, button } = van.tags;

const App = () => {
   const isOpen = van.state(false);
   const imgSrc = van.bind(isOpen, (v) => v ? "open-icon.png" : "close-icon.png");
   return div(
    { class: "App" },
    button({ onClick: () => isOpen.val = !isOpen.val; }),
   img({ src: imgSrc })
}

```

**Comment by Tao-VanJS** [comment, marked as answer]
Hi,

Thanks for reaching out!

The idiomatic solution for your use case is to use [State-derived properties](https://vanjs.org/tutorial#state-derived-prop). Thus instead of having:

```javascript
const imgSrc = van.bind(isOpen, (v) => v ? "open-icon.png" : "close-icon.png");
```

you can have:

```javascript
const imgSrc = {deps: [isOpen], f: (v) => v ? "open-icon.png" : "close-icon.png"}
```

This technique is illustrated in sample app [JSON Inspector](https://vanjs.org/demo#json-inspector) ([code](https://github.com/vanjs-org/vanjs-org.github.io/blob/master/code/json-inspector.html#L23)) and [Textarea with Autocomplete](https://vanjs.org/demo#auto-complete) ([code](https://github.com/vanjs-org/vanjs-org.github.io/blob/master/code/auto-complete-derived-props.ts#L26)).

btw: I guess `onClick` should be `onclick` which is a native property of DOM node.
